src/backend/api/meals.js

```diff
@@ -2,13 +2,81 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+router.get("/", async (req, res) => {
+  const { maxPrice, availableReservations, title, dateAfter, sortKey, sortDir, dateBefore, limit } =
+    req.query;
+  const response = {
+    data: [],
+    status: 200,
+    message: "ok",
+  };
+
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    if (maxPrice && !isNaN(maxPrice)) {
+      const result = await knex("meals").select("*").where("price", "<", +maxPrice);
+      response.data = result;
+    }
+
+    if (["meal_time", "max_reservation", "price"].includes(sortKey)) {
+      const result = await knex("meals")
+        .select("*")
+        .orderBy(sortKey, sortDir === "desc" ? "desc" : "asc");
+      response.data = result;
+    }
+
+    if (title) {
+      const result = await knex("meals").select("*").where("title", "like", `%${title}%`);
+      response.data = result;
+    }
+
+    if (availableReservations !== undefined) {
+      const isAvailable = availableReservations.toLowerCase() === "true";
+
+      if (isAvailable) {
+        const result = await knex("meals")
+          .countDistinct("reservation.id as total_reservations")
+          .leftJoin("reservation", "meals.id", "=", "reservation.meal_id")
+          .groupBy("meals.id", "meals.title", "meals.max_reservation")
+          .select("meals.title", "meals.max_reservation")
+          .having(function () {
+            this.sum("total_reservations", ">", "meals.max_reservation");
+          });
+        response.data = result;
+      } else {
+        const result = await knex("meals")
+          .countDistinct("reservation.id as total_reservations")
+          .leftJoin("reservation", "meals.id", "=", "reservation.meal_id")
+          .groupBy("meals.id", "meals.title", "meals.max_reservation")
+          .select("meals.title", "meals.max_reservation")
+          .having(function () {
+            this.sum("total_reservations", "<=", "meals.max_reservation");
+          });
+        response.data = result;
+      }
+    }
+
+    if (dateAfter) {
+      const result = await knex
+        .select("*")
+        .from("meals")
+        .where("meal_time", ">", new Date(dateAfter));
+      response.data = result;
+    }
+
+    if (dateBefore) {
+      const result = await knex("meals").select("*").where("meal_time", "<", dateBefore);
+      response.data = result;
+    }
+
+    if (limit && !isNaN(limit)) {
+      const result = await knex("meals").select("*").limit(parseInt(limit));
+      response.data = result;
+    }
+
+    res.status(response.status).json({ data: response.data, message: response.message });
```

Comment:
In the case that there are no query parameters provided, your endpoint returns an empty array, even though it should return all elements.